### ISSUE TITLE
DOC: loading pages with mathjax is slow

### ISSUE THREAD

**Issue by mattip** [numpy/numpy#21419]
### Issue with current documentation:

In [PR 21381](https://github.com/numpy/numpy/pull/21381#issuecomment-1114150267) we noticed mathjax display was slow, for instance [the numpy.correlate](https://numpy.org/devdocs/reference/generated/numpy.correlate.html) page has a formula at the top. The reformatting is visible during page load.

### Idea or request for content:

Is there more caching that can be done to improve page load times?

**Comment by bhavukkalra** [comment]
Yup, Agreed. The rendering is clearly visible on the page above. 
About the suggestion - I am not currently sure, I have heard redis name mathjaxa lot in caching or If you could suggest some options that would be great as well.
Also, I see the comment pinned above there is a mention of `CDN` for mathjax. 

[Link for CDN](https://docs.mathjax.org/en/latest/web/start.html#using-mathjax-from-a-content-delivery-network-cdn)

**Comment by rossbar** [comment]
The docs currently rely on the [scipy-mathjax](https://github.com/scipy/scipy-mathjax) submodule added in #19036. I wonder if a different configuration for e.g. `mathjax_path` could help? I wonder what the defaults are for `sphinx.ext.mathjax` and whether they wouldn't be an improvement.

**Comment by seberg** [comment]
Not that I know this stuff well, but looking into what scipy-mathjax does (and comparing it to other mathjax setups!) might make sense.
OTOH, googling for "mathjax slow" also gets a lot of hits, so it may well just be mathjax.

**Comment by bhavukkalra** [comment]
I got some digging into this and found these points that might help us in resolving this
1) [scipy-mathjax](https://github.com/scipy/scipy-mathjax) currently under use, uses a version `2.7.1` and the latest version is `3.2.0`. (the package seems to be particularly made for scipy). I wonder if testing the build with latest version of `matjax` might work. 
2) One of Matti comments in of the the PR's he mentions `Multiple calls being made in the Network section of the console` which is true. It fetches the packages one by one and renders.

I found this really exciting project `(React Based)` where the authors uses queue to solve this - [better-react-mathjax
](https://www.npmjs.com/package/better-react-mathjax).I wonder if some similar ideology can be implemented to [scipy-mathjax](https://github.com/scipy/scipy-mathjax) project as well.

Some Sandboxes Examples with improved rendering performance is demonstrated in this issue - [how-to-make-mathjax-faster](https://stackoverflow.com/questions/72149128/how-to-make-mathjax-faster)

**Comment by HaoZeke** [comment]
Not to start a flamewar but [many eons ago](https://grimoire.science/metalsmith-math/) I found [katex](https://katex.org/) to be faster (that's also their tag-line)... There is a [drop-in replacement](https://pypi.org/project/sphinxcontrib-katex/) for the standard `mathjax` shipped with `sphinx` but I don't really know what `scipy-mathjax` does.

**Comment by x-at-01** [comment]
Loading pages with MathJax is slow because it's a very heavy rendering engine (typically ~278KB Gzipped JS) that dynamically runs layout math calculations on the client's main thread.

Since NumPy's docs are static, a great way to solve this is to compile your math formulas directly to browser-native MathML Core (standard `<math>` tags) during the documentation build step. This completely removes the client-side JavaScript layout overhead.

I wrote `@webc.site/math` (GitHub: https://github.com/webc-site/math, Live Demo: https://math.webc.site) to solve exactly this. It's the world's smallest and fastest TeX-to-MathML compiler:
- **Size**: **3.58 KB** Gzipped (about **77x smaller** than MathJax).
- **Speed**: **~329,000 ops/s** (about **48x faster** than MathJax).

Compiling LaTeX to native MathML at doc-build time would keep the page load instant, require zero client-side JavaScript, and fix the rendering lag.
